src/ucp/wireup/select.c

```diff
@@ -1302,8 +1304,10 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
             }
 
             added_lanes += ucp_wireup_add_bw_lanes(select_params, &bw_info,
-                                                   context->mem_type_access_tls[mem_type],
+                                                   (~tl_bitmap &
+                                                    context->mem_type_access_tls[mem_type]),
                                                    UCP_NULL_LANE, select_ctx);
+            tl_bitmap   |= context->mem_type_access_tls[mem_type];
```

Comment:
what is the reason for this? why TLs of one memtype will exclude TLs of other memtype?
seems something else is needed here , and this code is somehow related to the order of memory types enum?

Reply:
Now, same TL(cuda_copy) is accessing two mem types (cuda and cuda_managed).  TL is excluded if it is already considered for other mem_type before.

Reply:
you mean because we don't need it again?
maybe do like this
needed_mem_types = ALL
while (needed_mem_types>0) {
   select lanes
   needed_mem_types &= ~supported_mem_types_by_selected_tl
}

Reply:
Yes. we dont need it again. we have overlapped TLs between two  mem_type's `contect->mem_type_access_tls[mem_type]`.  `cuda_copy` TL is common for both CUDA and CUDA_MANAGED. As we are looping over all mem_types, it is hitting assert inside ucp_wireup_add_bw_lanes if it is called twice with the same(overlap) tl_bitmap saying it is trying to add the same bw lane twice to lane descs.

Reply:
I'm not clear on your suggestion, will check offline with you.